activerecord/test/cases/transactions_test.rb

```diff
@@ -175,6 +175,13 @@ def topic.after_save_for_transaction
     assert topic.new_record?, "#{topic.inspect} should be new record"
   end
 
+  def test_transaction_rollback_retore_state_in_save_for_persited_records
```

Comment:
There's a typo in `restore` here. Can we also make this name clearer, I'm not sure what exactly this is testing from the name.